packages/contracts/src/contracts/anchor.ts

```diff
@@ -65,56 +63,58 @@ export class AnchorContract {
     await recipient.wait();
   }
 
-  private async getDepositEvents(commitment: string | null = null) {
-    const filter = this._contract.filters.Deposit(commitment, null, null);
+  private async getDepositLeaves(): Promise<string[]> {
+    const filter = this._contract.filters.Deposit(null, null, null);
 
     const currentBlock = await this.web3Provider.getBlockNumber();
-    const startingBlock = 1; // Read starting block from cached storage
+    const storedContractInfo = await this.mixersInfo.getMixerStorage(this._contract.address);
+
+    const startingBlock = storedContractInfo.lastQueriedBlock; // Read starting block from cached storage
     var logs = []; // Read the stored logs into this variable
 
     try {
       logs = await this.web3Provider.getLogs({
         fromBlock: startingBlock,
-        toBlock: 'latest',
+        toBlock: currentBlock,
         ...filter,
       });
     } catch (e) {
       console.log(e);
 
       // If there is a timeout, query the logs in block increments.
       if (e.code == -32603) {
-        for (var i=startingBlock; i < currentBlock; i+= 1000)
+        for (var i=startingBlock; i < currentBlock; i+= 50)
         {
           logs = [...logs, ...(await this.web3Provider.getLogs({
             fromBlock: i,
-            toBlock: (currentBlock - i > 1000) ? i + 1000 : currentBlock,
+            toBlock: (currentBlock - i > 50) ? i + 50 : currentBlock,
             ...filter,
           }))];
+
+          console.log(`Getting logs for block range: ${i} through ${i+50}`)
```

Comment:
This should be removed before integrating, but it shows the issue with EVM. This worked with an increment of 1000 around the time of beresheet v2 launch ~block 3000.

Reply:
Can you try to use the dedicated logger? I thought we had one in he dApp. You can keep it in but set it as a debug log.

Reply:
I put this log in the mixerLogger